utils/rollup-shader-chunks.mjs

```diff
@@ -0,0 +1,52 @@
+import { createFilter } from '@rollup/pluginutils';
+
+/** @typedef {import('rollup').Plugin} Plugin */
+/** @typedef {string | string[]} GlobPattern */
+/**
+ * @typedef {Object | null} PluginOptions
+ * @property {GlobPattern?} include - pattern(s array) to import
+ * @property {GlobPattern?} exclude - pattern(s array) to ignore
+ * @property {boolean?} enabled - enable the plugin
+ */
+
+/**
+ * @type {readonly string[]}
+ */
+const DEFAULT_SHADERS = Object.freeze(['**/*.js']);
+
+/**
+ * @param {PluginOptions} options - Plugin config object
+ * @returns {Plugin} The plugin that converts shader code.
+ */
+export function shaderChunks({
+    include = DEFAULT_SHADERS,
+    exclude = undefined,
+    enabled = true
+} = {}) {
```

Comment:
Thats fair still, all Rollup plugins are expected to implement this so they fit with global include and exclude. 

200ms added to build times. Chart published in issue.  Had to get 5 runs on both machines.

Reply:
> 200ms added to build times

Could we skip this for all debug builds? Faster build is more important than smaller file there. (unless we already skip it there)

Reply:
This line is doing that, the same as it did before.
https://github.com/playcanvas/engine/pull/5602/files#diff-47407fecafdf5f5cd55403c3de457833ddf9b6fab45253c04e1dc4c7cb4495b1R353

Reply:
> Thats fair still, all Rollup plugins are expected to implement this so they fit with global include and exclude. I'm following the plugin guidelines.

Global include and exclude doesn't even make sense and the wasted time for nothing is the best examples for that. Why even waste time profiling something that you know is just useless?

And if it would be expected, it wouldn't be optional... couldn't find that plugin guideline either you talk about, link?

Anyway, thank you for pointing out that the plugin was pratically disabled! Why not just fix the regex?

```js
    const filter = createFilter([
        '**/vert/*.js',
        '**/frag/*.js'
    ], []);
```

Reply:
Oops, I didn't read the first sentence, so I apologize for the over-engineering statement. Sensible and good work, thank you!